automation/patch_igz/deploy_remote.py

```diff
@@ -0,0 +1,314 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import io
+import logging
+import os
+import re
+import subprocess
+import sys
+from contextlib import contextmanager
+from typing import List
+
+import click
+import coloredlogs
+import paramiko
+import yaml
+
+log_level = logging.INFO
+fmt = "%(asctime)s %(levelname)s %(message)s"
+logging.basicConfig(level=log_level)
+logger = logging.getLogger("mlrun-deploy")
+coloredlogs.install(level=log_level, logger=logger, fmt=fmt)
+
+
+class MlRunDeployer(object):
+    class Consts(object):
+        mandatory_fields = {"DATA_NODES", "SSH_USER", "SSH_PASSWORD", "DOCKER_REGISTRY"}
+
+    def __init__(self, conf_file):
+        self._config = yaml.safe_load(conf_file)
+        self._validate_config()
+
+    def patch_mlrun_api(self):
+        branch = re.sub(r"\\|@|/|:|#|~|\+", "_", self._find_current_git_branch())
+        vers = self._get_current_version()
+
+        nodes = self._config["DATA_NODES"]
+        if not isinstance(nodes, list):
+            nodes = [nodes]
+
+        image_tag = self._get_image_tag(branch, vers)
+        self._make_mlrun_api(image_tag)
+        built_image = "{}/mlrun-api:{}".format(
+            self._config["DOCKER_REGISTRY"], image_tag
+        )
+
+        if self._username is not None:
+            self._exec_local(
+                [
+                    "docker",
+                    "login",
+                    "--username",
+                    self._username,
+                    "--password",
+                    self._password,
+                ],
+                live=True,
+            )
+
+        self._push_docker_image(built_image)
+
+        node = nodes[0]
+        self._connect_to_node(node)
+        try:
+            self._replace_deploy_policy()
+            self._replace_deployment_images(built_image)
+            self._wait_deployment_ready()
+        finally:
+            out = self._exec_remote(
+                [
+                    "kubectl",
+                    "-n",
+                    "default-tenant",
+                    "get",
+                    "pods",
+                ],
+            )
+
+            for line in out.splitlines():
+                if "mlrun-api" in line:
+                    logger.info(line)
+
+            self._disconnect_from_node()
+
+        logger.info(
+            "Deployed branch successfully! Yay! (Note this may not survive system restarts)"
+        )
+
+    def _validate_config(self):
+        missing_fields = self.Consts.mandatory_fields - set(self._config.keys())
+        if len(missing_fields) > 0:
+            raise RuntimeError(f"Mandatory options not defined: {missing_fields}")
+
+        self._username = self._config.get("REGISTRY_USERNAME")
+        self._password = self._config.get("REGISTRY_PASSWORD")
```

Comment:
```suggestion
        self._registry_username = self._config.get("REGISTRY_USERNAME")
        self._registry_password = self._config.get("REGISTRY_PASSWORD")
```
And move these to the init. Also why are these the only members you set from the config? What makes them special? I believe either set all config keys as members or none of them, or have a good reason to only set a subset of them.